### PR TITLE
Backport 1.3: fix bug in get_one_and_zeros_padding()

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
The function get_one_and_zeros_padding() accepts invalid padded data.

Straightforward backport of the original PR by @mickrau : #1236

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -48,6 +48,8 @@ Bugfix
    * Fix issue in RSA key generation program programs/x509/rsa_genkey
      where the failure of CTR DRBG initialization lead to freeing an
      RSA context without proper initialization beforehand.
+   * Fix bug in get_one_and_zeros_padding that sometimes accepted invalid
+     padding. Found and fixed by Micha Kraus.
```
Comment (hanno-arm) [inline ChangeLog]: It would be good to be more specific here; in particular, it should be noted that this padding is not used in SSL/TLS as far as I see.

Comment (hanno-arm) [reply ChangeLog]: For example, we might name the relevant padding identifier `{MBEDTLS/POLARSSL}_PADDING_ONE_AND_ZEROS` to be used in `[mbedtls_]cipher_set_padding_mode`.

**Comment by gilles-peskine-arm** [comment]
CI failed only on the IAR8 step which is not supported for 1.3 anyway, so ok to merge pending review.

File: ChangeLog (hunk 1 above)
Comment (gilles-peskine-arm) [reply ChangeLog]: How about this?

> Fix bug in cipher decryption with `MBEDTLS_PADDING_ONE_AND_ZEROS` that sometimes accepted invalid padding. (Not used in TLS.) Found and fixed by Micha Kraus.

Comment (hanno-arm) [reply ChangeLog]: @gilles-peskine-arm Sounds good to me, thanks!

Comment (gilles-peskine-arm) [reply ChangeLog]: @hanno-arm @Patater I forced-push a replacement for the last commit in the PR, with the changelog entry tweaked per the discussion here.

**Pull request merged**